Adding a more detailed pediatric cancer model investigation

gwaybio opened the pull request:
Also cleaning and rerunning the first data exploration notebook

gwaybio committed:
reorder and run data visualization, add more specific view of pediatric cancer

gwaybio committed:
rerun figure generation

d33bs commented on 1.data-exploration/figures/sample_cancer_types_bar_chart.png:
Due to the number of text labels, would it make sense to use a horizontal bar chart here (switching the x and y values)? Doing this might mean increased readability.

gwaybio replied:
yep! Will do

d33bs commented on 1.data-exploration/figures/age_categories_bar_chart.png:
Consider adding titles to this and other charts to help distinguish what they portray.

gwaybio replied:
Yep, this would be helpful. Adding!

d33bs commented on 1.data-exploration/1.model_data_visualization.ipynb:
```diff
@@ -1019,316 +960,258 @@
  "metadata": {
   "kernelspec": {
-   "display_name": "gene_dependency_representations",
+   "display_name": "Python 3 (ipykernel)",
```
Did this notebook use a new environment to run? Consider re-running within the `gene_dependency_representations` environment from `figure_environment.yml` if appropriate.

gwaybio replied:
oops! Thanks for calling this out. Will fix

d33bs commented on 1.data-exploration/scripts/nbconverted/1.model_data_visualization.py:
```diff
@@ -150,69 +148,116 @@
+    .primary_disease
+    .value_counts()
+    .sum()
+)
```
Noticing the repetition here, consider using [`pd.DataFrame.groupby().count()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html) combined with [`np.where()`](https://numpy.org/doc/stable/reference/generated/numpy.where.html) to help collect the data in a similar way and also open possibilities to create a data visualization (similar to comment for previous notebook cell). See [this SO post](https://stackoverflow.com/a/45083237) for more details on the approach.

gwaybio replied:
Thanks for pointing this out - I think given the amount of time to implement and the relatively low pay-off (we're simply using the print out and only running this once) I think I will skip to save time!

d33bs commented on 1.data-exploration/scripts/nbconverted/1.model_data_visualization.py:
```diff
@@ -150,69 +148,116 @@
-sgb_output = pathlib.Path("./figures/sample_gender_bar_chart.png")
-gendersamp_plot.save(sgb_output)
+# What is the distribution of pediatric tumor types
+pediatric_model_df.primary_disease.value_counts()
```
Would it be meaningful to export the displayed information to another visual in addition to the others provided in this notebook?

gwaybio replied:
yes! that is a good idea... I will add. Should be quick.

gwaybio replied:
yes! I added something in recent commit

d33bs commented on 1.data-exploration/scripts/nbconverted/1.model_data_visualization.py:
```diff
@@ -150,69 +148,116 @@
-    gg.ggplot(df_age_visual, gg.aes(x="age_distribution"))
-    + gg.geom_density()
-    + gg.geom_vline(xintercept=adult_threshold, linetype="dashed", color="red")
+pediatric_model_df = (
+    model_df
+    .query("age_categories == 'Pediatric'")
+    .query("DepMap_ID in @sample_overlap")
+    .reset_index(drop=True)
```
I don't see anything that makes me concerned about the formatting more generally, but the multi-line syntax in this notebook made me wonder about formatting tools which might be helpful if they're not already in use. Sharing along [`black[jupyter]`](https://black.readthedocs.io/en/stable/getting_started.html#installation) and [`jupyterlab_code_formatter`](https://github.com/ryantam626/jupyterlab_code_formatter), which can be helpful depending on your Jupyter IDE and development environment.

gwaybio replied:
yep, good point. I haven't ever done this before, but it's about time that I start!

I ended up going with jupyterlab_code_formatter, and made corresponding changes in the environment file.

d33bs reviewed (approved):
Nice work! I left a few comments and thoughts. Overall thought this LGTM! Please don't hesitate to let me know if you have any questions.

gwaybio committed:
autoformatting and responding to PR comments

changed orientation of figure, added titles, rerun with correct environment

gwaybio committed:
rerun notebook to generate all figures

update themes, rotation, adding new figure

gwaybio committed:
update environment

adding conda-forge namespace, adding jupyterlab_code_formatter and required packages, removing anaconda channel

gwaybio commented:
Thanks for the review @d33bs - I am happy with this PR and will be merging now.
